src/Inputs/ComboBox.tsx

```diff
@@ -0,0 +1,297 @@
+                    const selected = String(value) === val;
+                    return (
+                        <SelectItem
+                            {...child.props}
+                            active={selected}
+                            onClick={onSelect}
+                            key={val}
+                        />
+                    );}
+            }
+
+            else
+
+            {
+
+            if ((child && isValidElement(child)) && (myWindow.inputValue && (((child.props.children).toLowerCase()).search(myWindow.inputValue.toLowerCase()))> (-1) )) {
+                const val = child.props.value;
+                const selected = String(value) === val;
+                return (
+                    <SelectItem
+                        {...child.props}
+                        active={selected}
+                        onClick={onSelect}
+                        key={val}
+                    />
+                );}
+
+        }
+        
+        return null;} );
+
+    const onSelect = useCallback(
+        ({ currentTarget }): void => {
+            onChange({
+                target: {
+                    value: currentTarget.getAttribute('value'),
+                    name,
+                },
+            });
+            myWindow.inputValue=currentTarget.firstChild.nodeValue;
+        },
+        [name],
+    );
+
+    useEffect((): void | (() => void | undefined) => {
+        if (expanded) {
+            const listener = (): void => {
+                setExpanded(false);
+            };
+            const timer = window.setTimeout((): void => {
+                window.addEventListener('keydown', listener, { once: true });
+            }, 10);
+            const timer2 = window.setTimeout((): void => {
+                window.addEventListener('click', listener, { once: true });
+            }, 10);
+            return (): void => {
+                window.clearTimeout(timer);
+                window.clearTimeout(timer2);
+                window.removeEventListener('keydown', listener);
+                window.removeEventListener('click', listener);
+            };
+        }
+        return undefined;
+    }, [expanded]);
+
+    const handleChange=(event:any)=>{
+        myWindow.inputValue=event.target.value;
+        setExpanded(true);
+    }
+
+    return (
+        <LabelLayout {...props}>
+            <Container>
+
+                    <InputFragment value={myWindow.inputValue}
+                        onChange={e=>
+                            handleChange(e)
+                            } 
+                            onKeyDown={e=>
+                                handleChange(e)
+                                } 
+                    onClick={e=>
+                        handleChange(e)
+                        } 
+                        
+                     />
+
+                {mount && (
+                    <SelectList
+                        limit={Math.min(3, limit)}

```

Comment:
3 is a magic number, use a constant NUM_VISIBLE_SELECTIONS = 3;

Reply:
done